arch/xtensa/src/esp32/esp32_spi.c

```diff
@@ -924,6 +955,21 @@ static void esp32_spi_dma_exchange(FAR struct esp32_spi_priv_s *priv,
       tp += n;
       rp += n;
     }
+
+#ifdef CONFIG_XTENSA_USE_SEPERATE_IMEM
+  if (esp32_ptr_extram(rxbuffer))
+    {
+      memcpy(rxbuffer, rp, bytes);
+      up_imm_free(rp);
```

Comment:
tp and rp are modified above. (tp += n)
i guess you should free the original pointer.

Reply:
Indeed.  Thanks, done.